src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,452 @@
+        } catch (IndexOutOfBoundsException error) {
+            String errorMessage = "Out of bounds value provided.";
+            String errorMessageEcho = String.format("List only has %d items, you entered %d!",
+                    userList.size(), taskNumber);
+            printMessage(errorMessage);
+            printMessage(errorMessageEcho);
+        } finally {
+            separator();
+        }
+    }
+
+    public static void loadTaskList(String fileContent) {
+        String[] lines = fileContent.split("\n");
+        for (String line : lines) {
+            String[] taskInfo = line.split("\\|");
+            switch (taskInfo[0]) {
+            case "T":
+                Todo todo = new Todo(taskInfo[2]);
+                if (taskInfo[1].equals("X")) {
+                    todo.setAsDone();
+                } else {
+                    todo.setAsNotDone();
+                }
+                userList.add(todo);
+                break;
+            case "D":
+                Deadline deadline = new Deadline(taskInfo[2], taskInfo[3]);
+                if (taskInfo[1].equals("X")) {
+                    deadline.setAsDone();
+                } else {
+                    deadline.setAsNotDone();
+                }
+                userList.add(deadline);
+                break;
+            case "E":
+                Event event = new Event(taskInfo[2], taskInfo[4], taskInfo[3]);
+                if (taskInfo[1].equals("X")) {
+                    event.setAsDone();
+                } else {
+                    event.setAsNotDone();
+                }
+                userList.add(event);
+                break;
+            }
+        }
+    }
+
+    public static void startDuke() throws IOException {
+        separator();
+        printMessage("Hello! I'm Duke");
+        printMessage("Let me check the current list of tasks");
+
+        // Check for database file
+        // Create parent directory and file if it does not exist,
+        // otherwise show current list of tasks
+        Path filepath = Paths.get("data", "duke.txt");
+        File databaseFile = filepath.toFile();
+        if (!databaseFile.exists()) {
+            printMessage("Database file does not exist, creating one now!");
+            databaseFile.getParentFile().mkdirs();
+            databaseFile.createNewFile();
+        } else {
+            printMessage("Loading previous task list...");
+            FileReader fileReader = new FileReader(databaseFile);
+            BufferedReader bufferedReader = new BufferedReader(fileReader);
+            String fileContent = "";
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                fileContent += line + "\n";
+            }
+
+            loadTaskList(fileContent);
+            printMessage("Done loading previous task list:");
+            String outputMessage = String.format("Loaded a total of %d tasks!", userList.size());
```

Comment:
perhaps consider extracting related sub-behaviours such as this section into another method to simplify readability of start duke.